segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DataFrameReader.java

```diff
@@ -105,12 +105,13 @@ public DataFrameRecord<T> getNext() throws DataCorruptionException, DurableDataL
                     DataFrameRecord.RecordInfo recordInfo = this.dataFrameInputStream.endRecord();
                     long seqNo = logItem.getSequenceNumber();
                     if (seqNo <= this.lastReadSequenceNumber) {
-                        throw new DataCorruptionException(String.format("Invalid Operation Sequence Number. Expected: larger than %d, found: %d.",
-                                this.lastReadSequenceNumber, seqNo));
+                        // In case there is a duplicate entry, we can safely discard it (see https://github.com/pravega/pravega/issues/6444)
+                        log.warn("Invalid Operation Sequence Number. Expected: larger than {}, found: {}. Discarding entry.",
```

Comment:
Is there a way to actually verify if both entries are indeed same (same payload not not just the seqNo) ?

Reply:
We depart from the assumptions that i) only one `BookkeeperLog` writes to its own ledgers (this is guaranteed by fencing), ii) the `OperationProcessor` keeps the strict monotonically increasing sequence numbers of operations (this is  checked in `OperationProcessor` and `SreamSegmentContainerMetadata`).

Assuming that, there could be other scenarios: i) there are duplicate entries but one of them is partially written or corrupted (in this case, the CRC check in Bookkeeper may notice an integrity problem in the entry contents), ii) The entry is correctly written, but is unrelated to Pravega (this would trigger a deserialization error during recovery).

Reply:
We depart from the assumptions that i) only one `BookkeeperLog` writes to its own ledgers (this is guaranteed by fencing), ii) the `OperationProcessor` keeps the strict monotonically increasing sequence numbers of operations (this is  checked in `OperationProcessor` and `SreamSegmentContainerMetadata`).

Assuming that, there could be other scenarios: i) there are duplicate entries but one of them is partially written or corrupted (in this case, the CRC check in Bookkeeper may notice an integrity problem in the entry contents), ii) The entry is correctly written, but is unrelated to Pravega (this would trigger a deserialization error during recovery).

Reply:
@sachin-j-joshi I have added verification of internal info of the DataFrame to have certainty (beyond the sequence number) that the entry is actually the same.